cli/command/environment.go

```diff
@@ -166,6 +172,60 @@ func (e *EnvironmentCommand) create(c *cli.Context) error {
 }
 
 func (e *EnvironmentCommand) delete(c *cli.Context) error {
+	if c.Bool("recursive") {
+		args, err := extractArgs(c.Args(), "NAME")
+		if err != nil {
+			return err
+		}
+
+		environmentID, err := e.CommandBase.resolveSingleEntityIDHelper("environment", args["NAME"])
+		if err != nil {
+			return err
+		}
+
+		// Delete Load Balancers
+		loadBalancers, err := e.client.ListLoadBalancers()
+		if err != nil {
+			return err
+		}
+
+		for _, loadBalancer := range loadBalancers {
+			if loadBalancer.EnvironmentID == environmentID {
+				if _, err := e.client.DeleteLoadBalancer(loadBalancer.LoadBalancerID); err != nil {
+					return err
+				}
+			}
+		}
+
+		// Delete Tasks
+		tasks, err := e.client.ListTasks()
+		if err != nil {
+			return err
+		}
+
+		for _, task := range tasks {
+			if task.EnvironmentID == environmentID {
+				if _, err := e.client.DeleteTask(task.TaskID); err != nil {
+					return err
+				}
+			}
+		}
+
+		// Delete Services
+		services, err := e.client.ListServices()
+		if err != nil {
+			return err
+		}
+
+		for _, service := range services {
+			if service.EnvironmentID == environmentID {
+				if _, err := e.client.DeleteTask(service.ServiceID); err != nil {
```

Comment:
should be `DeleteService` (this is an example why it is important to test these code paths in the unit tests).